minichlink: init at version bbe0e83

cyber-murmel opened the pull request:
currently there is no explicit versioning, as it is part of a bigger repo

###### Description of changes
this add the ch32v003 debugging tool by cnlohr

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I managed to flash the blink program to the IC using this.

evanrichter commented on maintainers/maintainer-list.nix:
```diff
@@ -10090,6 +10090,13 @@
     githubId = 1651325;
     name = "maralorn";
   };
+  marble = {
```
It's more common to add yourself as a maintainer in one commit, then provide package changes (or new packages) in the next commit

evanrichter commented on pkgs/development/embedded/minichlink/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+stdenv.mkDerivation rec {
+  pname = "minichlink";
+  version = "bbe0e83";
```
No harm using the full sha1 git hash

cyber-murmel committed:
maintainers: add marble

cyber-murmel committed:
minichlink: init at version bbe0e83

this add the ch32v003 debugging tool by cnlohr
currently there is no explicit versioning, as it is part of a bigger repo

evanrichter commented on pkgs/top-level/all-packages.nix:
```diff
@@ -18855,6 +18855,8 @@ with pkgs;
 
   md-tangle = callPackage ../development/tools/literate-programming/md-tangle { };
 
+  minichlink = callPackage ../development/embedded/minichlink { };
```
these lines are in the first maintainer commit, should be moved to the second

donovanglover commented:
If you are still interested in this PR, please review [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) since a lot has changed. Otherwise feel free to close it.
